### PR TITLE
[DOCS] Add examples of using named outputs in extensions

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Add examples of using named outputs in extensions*

### Tickets:
 - *108007*

### CODE REVIEW

File: docs/Extensibility_UG/frontend_extensions.md [hunk 1]
```diff
@@ -381,6 +381,21 @@ To access original framework operation attribute value and connect to inputs, ``
 The conversion function should return a vector of node outputs that are mapped to 
 corresponding outputs of the original framework operation in the same order.
 
+Some frameworks require names of outputs of operation to be provided during conversion.
+For paddlepaddle operations it is generally necessary to provide names for all outputs using ``NamedOutputs`` container.
+The next example shows such conversion for ``top_k_v2`` operation.
```
Comment (tsavina) [inline docs/Extensibility_UG/frontend_extensions.md:386]:
```suggestion
Some frameworks require output names of the operation to be provided during conversion.
For PaddlePaddle operations, it is generally necessary to provide names for all outputs using the ``NamedOutputs`` container.
The next example shows such conversion for the ``top_k_v2`` operation.
```

File: docs/Extensibility_UG/frontend_extensions.md [hunk 2]
```diff
@@ -381,6 +381,21 @@ To access original framework operation attribute value and connect to inputs, ``
...
+
+.. doxygensnippet:: docs/snippets/ov_extensions.cpp
+   :language: cpp
+   :fragment: [frontend_extension_paddle_TopK]
+
+For tensorflow framework if operation has more then one output it is recommended to assign names to those outputs using ``NamedOutputVector`` structure which allows both indexed and named output access.
+The next example shows such conversion for ``TopKV2`` operation.
```
Comment (tsavina) [inline docs/Extensibility_UG/frontend_extensions.md:393]:
```suggestion
For TensorFlow framework, if an operation has more than one output, it is recommended to assign names to those outputs using the``NamedOutputVector`` structure which allows both indexed and named output access.
The next example shows such conversion for the ``TopKV2`` operation.
```

File: docs/Extensibility_UG/frontend_extensions.md [hunk 3]
```diff
@@ -381,6 +381,21 @@ To access original framework operation attribute value and connect to inputs, ``
...
 
+Some frameworks require output names of the operation to be provided during conversion.
+For PaddlePaddle operations, it is generally necessary to provide names for all outputs using the ``NamedOutputs`` container.
+The next example shows such conversion for the ``top_k_v2`` operation.
+
+.. doxygensnippet:: docs/snippets/ov_extensions.cpp
+   :language: cpp
+   :fragment: [frontend_extension_paddle_TopK]
+
+For TensorFlow framework, if an operation has more than one output, it is recommended to assign names to those outputs using the``NamedOutputVector`` structure which allows both indexed and named output access.
```
Comment (mvafin) [inline docs/Extensibility_UG/frontend_extensions.md:392]:
```suggestion
For TensorFlow framework, if an operation has more than one output, it is recommended to assign names to those
outputs using the ``NamedOutputVector`` structure which allows both indexed and named output access.
```

File: docs/Extensibility_UG/frontend_extensions.md [hunk 4]
```diff
@@ -381,6 +381,22 @@ To access original framework operation attribute value and connect to inputs, ``
...
+The next example shows such conversion for the ``top_k_v2`` operation.
+
+.. doxygensnippet:: docs/snippets/ov_extensions.cpp
+   :language: cpp
+   :fragment: [frontend_extension_paddle_TopK]
+
+For TensorFlow framework, if an operation has more than one output, it is recommended to assign names to those
+outputs using the ``NamedOutputVector`` structure which allows both indexed and named output access.
+The next example shows such conversion for the ``TopKV2`` operation.
+
```
[the remark this answers is not in the document]
Comment (mvafin) [reply docs/Extensibility_UG/frontend_extensions.md:398]: Added

File: docs/snippets/ov_extensions.cpp [hunk 5]
```diff
@@ -204,13 +206,49 @@ auto extension = std::make_shared<ov::frontend::OpExtension<CustomElu>>("aten::e
...
+//! [frontend_extension_paddle_TopK]
+
+//! [frontend_extension_tf_TopK]
+core.add_extension(ov::frontend::ConversionExtension("TopKV2", [](const ov::frontend::NodeContext& node) {
+    auto input = node.get_input(0);
+    auto k_input = node.get_input(1);
+    bool sorted = node.get_attribute<bool>("sorted", true);    
+    auto mode = ov::opset11::TopK::Mode::MAX;
+    auto sort_type = sorted ? ov::opset11::TopK::SortType::SORT_VALUES : ov::opset11::TopK::SortType::SORT_INDICES;
+    auto top_k = std::make_shared<ov::opset11::TopK>(input, k_input, -1, mode, sort_type, ov::element::i32, true);
```
Comment (rkazants) [inline docs/snippets/ov_extensions.cpp:247]: also, propose to set the friendly name to preserve the node name

**Review by rkazants — APPROVED** [review]
Last comment is remained. The other looks good to me

File: docs/snippets/ov_extensions.cpp (hunk 5 above)
Comment (mvafin) [reply docs/snippets/ov_extensions.cpp:247]: For the purpose of simplifying the snippet code I removed friendly names, the purpose of this snippet is not to show what ideal extension should be, but to provide a working example of named ports usage.

File: docs/Extensibility_UG/frontend_extensions.md [hunk 6]
```diff
@@ -381,6 +381,25 @@ To access original framework operation attribute value and connect to inputs, ``
...
+The next example shows such conversion for the ``top_k_v2`` operation.
+
+.. doxygensnippet:: docs/snippets/ov_extensions.cpp
+   :language: cpp
+   :fragment: [frontend_extension_paddle_TopK]
+
+For TensorFlow framework, if an operation has more than one output, it is recommended to assign names to
+those outputs using the ``NamedOutputVector`` structure which allows both indexed and named output access.
+For description of TensorFlow operations where names of outputs are also listed please refer to
+[tf.raw_ops](https://www.tensorflow.org/api_docs/python/tf/raw_ops/) documentation page.
```
Comment (tsavina) [inline docs/Extensibility_UG/frontend_extensions.md:396]:
```suggestion
For a description of TensorFlow operations, including the names of their outputs, refer to the
`tf.raw_ops <https://www.tensorflow.org/api_docs/python/tf/raw_ops/>`__ documentation page.
```

**Review by tsavina — APPROVED** [review]

**Pull request merged**